examples/test/post-processing/components/bloom.js

```diff
@@ -0,0 +1,161 @@
+
+  schema: {
+    passes: { default: '0 1 2 3 4 5 7' },
+    threxp: { type: 'vec2', default: new THREE.Vector2(16, 0) },
+    intensity: { type: 'float', default: 1.0 }
+  },
+
+  setupPostState: function () {
+    this.scenePost = new THREE.Scene();
+    this.cameraPost = new THREE.OrthographicCamera(-1, 1, 1, -1, 0, 1);
+    this.quadPost = new THREE.Mesh(new THREE.PlaneBufferGeometry(2, 2), null);
+    this.scenePost.add(this.quadPost);
+  },
+
+  init: function () {
+    this.setupPostState();
+    this.brightMaterial = new THREE.ShaderMaterial({
+      uniforms: {
+        srcTexture: { type: 't', value: null },
+        threxp: { type: 'v2', value: new THREE.Vector2(0, 0) }
+      },
+      vertexShader: this.vertexBrightText,
+      fragmentShader: this.fragmentBrightText,
+
+      depthWrite: false
+
+    });
+    this.blurMaterial = new THREE.ShaderMaterial({
+      uniforms: {
+        iteration: { type: 'f', value: 0 },
+        srcTexture: { type: 't', value: null },
+        pixelSize: { type: 'v2', value: new THREE.Vector2(0, 0) }
+      },
+      vertexShader: this.vertexBlurText,
+      fragmentShader: this.fragmentBlurText,
+
+      depthWrite: false
+
+    });
+    this.quadPost.material = this.brightMaterial;
+    var rtOptions = { minFilter: THREE.LinearFilter, magFilter: THREE.NearestFilter, format: THREE.RGBAFormat };
+    this.renderTargets = [
+      new THREE.WebGLRenderTarget(1, 1, rtOptions),
+      new THREE.WebGLRenderTarget(1, 1, rtOptions)
+    ];
+    this.el.emit('post-modified');
+  },
+
+  remove: function () {
+    this.el.emit('post-modified');
+  },
+
+  update: function (oldData) {
+    this.postIntensity = this.data.intensity;
+    console.log(this.postIntensity);
+  },
+
+  // We're doing bloom at half resolution for performance. Most blurry effects can get away with that.
+  getTargets: function () {
+    var srt = this.el.sceneEl.renderTarget;
+    for (var i = 0; i < 2; i++) {
+      var rt = this.renderTargets[i];
+      if (rt.width !== srt.width / 2 || rt.height !== srt.height / 2) {
+        rt.setSize(srt.width / 2, srt.height / 2);
+      }
+    }
+    return this.renderTargets;
+  },
+
+  tock: function () {
+    var rts = this.getTargets();
+    var el = this.el;
+    var scene = el.sceneEl;
+    var uns = this.brightMaterial.uniforms;
+    var rt = scene.renderTarget;
+    // Bright pass: isolate and highlight the brightest parts of the scene.
+    this.scenePost.overrideMaterial = this.brightMaterial;
+    uns.threxp.value = this.data.threxp;
+    uns.srcTexture.value = rt;
+    rt = rts[0]; // Output to the first of the ping pong targets.
+    scene.renderer.render(this.scenePost, this.cameraPost, rt);
+
+    // Ping pong the render targets progressively blurring more on each pass.
+    uns = this.blurMaterial.uniforms;
+    this.scenePost.overrideMaterial = this.blurMaterial;
+    var passes = this.data.passes.split(' ');
+    for (var i = 0; i < passes.length; i++) {
```

Comment:
we could but kawase bloom is supposed to be tunable with these params. Using different not linearly incrementing values would result in a different kind of bloom, check the intel article which explains it. In practice it doesn't give very good results for small blur kernels, even though it's very fast with large ones. A standard sepparable gaussian would be more straightforward, should I revert to that? https://software.intel.com/en-us/blogs/2014/07/15/an-investigation-of-fast-real-time-gpu-based-image-blur-algorithms